static/src/javascripts/projects/commercial/modules/dfp/display-lazy-ads.ts

```diff
@@ -0,0 +1,30 @@
+import { dfpEnv } from './dfp-env';
+import { enableLazyLoad } from './lazy-load';
+import { loadAdvert } from './load-advert';
+
+const advertsToInstantlyLoad = ['dfp-ad--im'];
+
+const instantLoad = () => {
+	const instantLoadAdverts = dfpEnv.advertsToLoad.filter((advert) =>
+		advertsToInstantlyLoad.includes(advert.id),
+	);
+
+	dfpEnv.advertsToLoad = dfpEnv.advertsToLoad.filter(
+		(advert) => !advertsToInstantlyLoad.includes(advert.id),
+	);
+
+	instantLoadAdverts.forEach(loadAdvert);
+};
+
+const displayLazyAds = (): void => {
+	window.googletag?.pubads().collapseEmptyDivs();
+	window.googletag?.enableServices();
+
+	instantLoad();
+
+	dfpEnv.advertsToLoad.forEach((advert) => {
+		enableLazyLoad(advert);
+	});
```

Comment:
Seems a bit odd to iterate over `dfpEnv.advertsToLoad` 3 separate times.

Do you find the following easier to parse? And then we get rid of `instantLoad` altogether.

```suggestion
	dfpEnv.advertsToLoad.forEach((advert) => {
	    if(advertsToInstantlyLoad.includes(advert.id))
	        loadAdvert(advert);
	    else
		    enableLazyLoad(advert);
	});
```

Reply:
I agree there is some unneccessary looping here.

The only problem being it looks like line 12 is setting `dfpEnv.advertsToLoad`.

I'll take a look at simplifying thanks

Reply:
If you pass the index as the second param of `forEach`, you can do `dfpEnv.advertsToLoad.splice(i, 1)`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice

Reply:
Side effects are hard to reason about. Why do we need to modify `dfpEnv`? So many questions!

Reply:
Yes working why this side effect is there is a larger task... in fact what all the state in `dfpEnv` is for at all.

I've refactored to reduce the loops from 4 to 3 for now.